script.js

```diff
@@ -1,4 +1,435 @@
+// set global variables
+// to track games and progress
+var numGamesDraw = 0;
+var numGamesWon = 0;
+var numGamesPlayed = 0;
+
+// to set game mode
+var currentGameMode = "enter user name";
+console.log(currentGameMode);
+
+// to set user name
+var userName = "";
+
+// for Korean game
+var lastWinner = 0;
+
+//set function to return percentage win
+var percentageWin = function () {
+  var winRate = Math.floor((numGamesWon / numGamesPlayed) * 100);
+  return winRate;
+};
+
+// function to generate random computer move
+var randomItem = function () {
+  var random = Math.floor(Math.random() * 3);
+  var programMove = "stone";
+  if (random == 1) {
+    programMove = "scissors";
+  }
+  if (random == 2) {
+    programMove = "paper";
+  }
+  return programMove;
+};
+
+// function to determine when game is a draw where computer's move and players move is the same.
+var drawGame = function (input, computerMove) {
+  return input == computerMove;
+};
+
+// function to determine when player wins in normal mode
+var playerWin = function (input, computerMove) {
+  return (
+    (input == "scissors" && computerMove == "paper") ||
+    (input == "stone" && computerMove == "scissors") ||
+    (input == "paper" && computerMove == "stone")
+  );
+};
+
+// function to determine when player wins in reversed mode
+// created another var so that we do not need to change the number of games and won percentage logic etc
+var playerWinReserved = function (input, computerMove) {
+  return (
+    (input == "stone" && computerMove == "paper") ||
+    (input == "paper" && computerMove == "scissors") ||
+    (input == "scissors" && computerMove == "stone")
+  );
+};
+
+// function to determine when player inputs an invalid guess.
+var invalid = function (input) {
+  return input != "scissors" && input != "stone" && input != "paper";
+};
+
+// function to determine when player loses (this is for Korean mode)
+var playerLose = function (input, computerMove) {
+  return (
+    (input == "stone" && computerMove == "paper") ||
+    (input == "paper" && computerMove == "scissors") ||
+    (input == "scissors" && computerMove == "stone")
+  );
+};
+
+// function to return the logo corresponding to the move.
+var iconLogo = function (look) {
+  var logo = "none";
+  if (look == "scissors" || look == "reversed scrissors") {
+    logo = "✂️";
+  }
+  if (look == "paper" || look == "reversed paper") {
+    logo = "📜";
+  }
+  if (look == "stone" || look == "reversed stone") {
+    logo = "🗿";
+  }
+  return logo;
+};
```

Comment:
SCREAM_CASE variables prevent propensity toward spelling errors: "reversed scrissors"

But this is a nice touch to the program 👍